agent/agent.go

```diff
@@ -318,3 +327,28 @@ func initializeStateManager(cfg *config.Config, taskEngine engine.TaskEngine, cl
 	}
 	return stateManager, nil
 }
+
+// initializeENIWatcher wraps up the setup for creating the ENI Watcher
+func initializeENIWatcher(ctx context.Context) (*eniWatcher.UdevWatcher, error) {
```

Comment:
`agent.go` is not a good example to follow.  I've been wanting to refactor it for a long time.